### PR TITLE
Default to SW mode when accessing app as PWA

### LINKED ISSUE #808: Make Service Worker the default if the app is being accessed as a PWA
This would be a step towards #196. It would be a very short PR (probably one-liner). It makes no sense that accessing the app as a PWA -- which requires Service Worker by definition -- opens the app in JQuery mode. So, the idea here is that if the app is being accessed from http(s): and Service Worker is available, it should open in SW mode by default (the user can still change back to JQuery).

This would not yet affect our extensions, which would open in JQuery mode by default.

This would also fix #801 implicitly. I'd recommend this being part of release 3.3.

### PR DESCRIPTION
**Pull request by Jaifroid**
Fixes #808 and #801. Basically a one-liner, but formatted as two for legibility 😊.
EDIT: It will be interesting to see what effect this has on the secure browser tests...

### CODE REVIEW

**Comment by Jaifroid** [comment]
Interesting that all tests passed. I was expecting the secure browser tests to fail due to differences between jQuery and SW mode, but clearly the differences are immaterial. That's good.

File: www/js/app.js [hunk 1]
```diff
@@ -93,7 +93,9 @@ define(['jquery', 'zimArchiveLoader', 'uiUtil', 'settingsStore','abstractFilesys
     // A parameter to access the URL of any extension that this app was launched from
     params['referrerExtensionURL'] = settingsStore.getItem('referrerExtensionURL');
     // A parameter to set the content injection mode ('jquery' or 'serviceworker') used by this app
-    params['contentInjectionMode'] = settingsStore.getItem('contentInjectionMode') || 'jquery'; // Defaults to jquery for now
+    params['contentInjectionMode'] = settingsStore.getItem('contentInjectionMode') || 
+        // Defaults to jquery in extensions, and serviceworker if accessing as a PWA
+        (/http/i.test(window.location.protocol) && isServiceWorkerAvailable()) ? 'serviceworker' : 'jquery';
```
Comment (mossroy) [inline www/js/app.js:98]: I would suggest to use a more specific regexp, like: 
```suggestion
        (/^https?$/i.test(window.location.protocol) && isServiceWorkerAvailable()) ? 'serviceworker' : 'jquery';
```
It should avoid matching a protocol like `this-is-not-http-protocol` (which does not exist, of course, it's an example)
NB : I did not test

**Review by mossroy — COMMENTED** [review]
Good idea, even if directly accessing the PWA is not the main usage

**Comment by mossroy** [comment]
> Interesting that all tests passed. I was expecting the secure browser tests to fail due to differences between jQuery and SW mode, but clearly the differences are immaterial. That's good.

That's very interesting.
Do we have a way to check which mode has been used by each browser from the UI tests logs?
Based on https://caniuse.com/mdn-api_serviceworker, it should use jQuery mode for Edge 15, and SW mode for all other ones (as IE11 does not run these tests at all)

**Comment by Jaifroid** [comment]
> That's very interesting. Do we have a way to check which mode has been used by each browser from the UI tests logs? Based on https://caniuse.com/mdn-api_serviceworker, it should use jQuery mode for Edge 15, and SW mode for all other ones (as IE11 does not run these tests at all)

I think it might depend on whether we can see the console output in the tests snapshots. If they are running in a secure context and the browser is capable of supporting SW mode, they should launch in SW mode with this PR.

File: www/js/app.js (hunk 1 above)
Comment (Jaifroid) [reply www/js/app.js:98]: OK, will do.

**Comment by Jaifroid** [comment]
So I guess this is ready to merge. I've tested on Firefox & Edge Chromium from localhost. You have to reset the app to see what it defaults to when there is no `contentinjectionmode` key in the localStorage. In both browsers it defaults to SW mode. In IE11 of course it remains in jQuery mode after a reset.

**Review by mossroy — APPROVED** [review]
OK for me

**Pull request merged**